[Merged by Bors] - devcontainers, devnet, ci, benchmarks via reproducible cacheble nix builds v1 (#1260)
ComposableFi/composable: .devcontainer/nix/README.md

# Overview

> The Nix based devcontainer. Is maintained to be the same as CI and DevNet

Reviewer: ```suggestion
The Nix based devcontainer. Has the same environment as available in CI and on devnet
```
